core/modules.rs

```diff
@@ -398,84 +398,46 @@ enum SymbolicModule {
   Mod(ModuleId),
 }
 
-#[derive(Default)]
-/// Alias-able module name map
-struct ModuleNameMap {
-  inner: HashMap<String, SymbolicModule>,
-}
-
-impl ModuleNameMap {
-  pub fn new() -> Self {
-    ModuleNameMap {
-      inner: HashMap::new(),
-    }
-  }
-
-  /// Get the id of a module.
-  /// If this module is internally represented as an alias,
-  /// follow the alias chain to get the final module id.
-  pub fn get(&self, name: &str) -> Option<ModuleId> {
-    let mut mod_name = name;
-    loop {
-      let symbolic_module = self.inner.get(mod_name)?;
-      match symbolic_module {
-        SymbolicModule::Alias(target) => {
-          mod_name = target;
-        }
-        SymbolicModule::Mod(mod_id) => return Some(*mod_id),
-      }
-    }
-  }
-
-  /// Insert a name associated module id.
-  pub fn insert(&mut self, name: String, id: ModuleId) {
-    self.inner.insert(name, SymbolicModule::Mod(id));
-  }
-
-  /// Create an alias to another module.
-  pub fn alias(&mut self, name: String, target: String) {
-    self.inner.insert(name, SymbolicModule::Alias(target));
-  }
-
-  /// Check if a name is an alias to another module.
-  #[cfg(test)]
-  pub fn is_alias(&self, name: &str) -> bool {
-    let cond = self.inner.get(name);
-    matches!(cond, Some(SymbolicModule::Alias(_)))
-  }
-}
-
 /// A collection of JS modules.
 #[derive(Default)]
-pub struct Modules {
+pub struct ModuleMap {
   ids_by_handle: HashMap<v8::Global<v8::Module>, ModuleId>,
   handles_by_id: HashMap<ModuleId, v8::Global<v8::Module>>,
   info: HashMap<ModuleId, ModuleInfo>,
-  by_name: ModuleNameMap,
+  by_name: HashMap<String, SymbolicModule>,
   next_module_id: ModuleId,
 }
 
-impl Modules {
-  pub fn new() -> Modules {
+impl ModuleMap {
+  pub fn new() -> ModuleMap {
     Self {
       handles_by_id: HashMap::new(),
       ids_by_handle: HashMap::new(),
       info: HashMap::new(),
-      by_name: ModuleNameMap::new(),
+      by_name: HashMap::new(),
       next_module_id: 1,
     }
   }
 
   pub fn get_id(&self, name: &str) -> Option<ModuleId> {
-    self.by_name.get(name)
+    let mut mod_name = name;
+    loop {
+      let symbolic_module = self.by_name.get(mod_name)?;
+      match symbolic_module {
+        SymbolicModule::Alias(target) => {
+          mod_name = target;
+        }
+        SymbolicModule::Mod(mod_id) => return Some(*mod_id),
+      }
+    }
```

Comment:
Maybe there should be a counter here to detect infinite loops?

doc string would be helpful for this function.

Reply:
I don't think so, this function hasn't changed in a long time and works as expected. Added a docstring.